detailed_docs/Wiki/0249_capabilities.md

```diff
@@ -0,0 +1,135 @@
+**And** SDL uses this new data set for apps/SDL Core communication until ignition cycle is performed.
+
+---
+
+#### Scenario 5:  SDL continues launching sequence if capabilities sent fom HMI are already present in `HMICapabilitiesCacheFile` file
+
+**Given**  Receiving HMI response for any of the UI/VR/TTS/Buttons/VehicleInfo/Navigation/RC capabilities
+
+**When** `HMICapabilitiesCacheFile` file already has the data set received in HMI response
+
+**Then** SDL does not overwrite the response to `HMICapabilitiesCacheFile` file  
+**And** SDL does not send corresponding request to get HMI capabilities to HMI for subsequent ignition cycles  
+**And** SDL follows the rest of the launch sequence
+
+---
+
+#### Scenario 6: SDL overwrites contents of received Dynamic RPC in hmi_capabilities_cache.json file
+
+**Given** Mobile application already obtained HMI capabilities in the ignition cycle
+
+**When** HMI sends a Dynamic RPC with changes to capabilities
+
+**Then** Core must overwrite contents of received notifications in `HMICapabilitiesCacheFile` file and use this data until ignition cycle is performed
+
+**_Dynamic RPCs:_**
+  * UI.OnLanguageChange
+  * VR.OnLanguageChange
+  * TTS.OnLanguageChange
+
+---
+
+#### Scenario 7: SDL deletes `HMICapabilitiesCacheFile` file when a master reset is performed
+
+**Given** All HMI Capabilities are presented in `HMICapabilitiesCacheFile` 
+
+**When** HMI sends OnExitAllApplications with reason `MASTER_RESET`
+
+**Then** SDL Core must delete `HMICapabilitiesCacheFile` in AppStorageFolder in [smartDeviceLink.ini](https://github.com/smartdevicelink/sdl_core/blob/master/src/appMain/smartDeviceLink.ini)
+
+## Feature 2: As a mobile application, I want to get actual HMI capabilities from SDL after HMI SW update
+
+### Scenario 1: SDL regenerates `HMICapabilitiesCacheFile` file when system SW version changes
+
+**Given** App sends RAI after HMI SW update 
+
+**When** SDL's local `ccpu_version` does not match with received from BC.GetSystemInfo(`ccpu_version`)
+
+**Then** SDL must delete `HMICapabilitiesCacheFile`  
+**And** SDL must send `GetCapabilities` requests for each interface (VR, TTS, UI, etc) to HMI and persist the responses received from HMI to `HMICapabilitiesCacheFile`  
+**And** SDL must set `IsHMICooperating` to true when HMI has responded to all the HMI capabilities requests  
+**And** SDL must respond to all pending RAI requests from mobile applications
+
+---
+
+### Scenario 2: SDL does not persist capabilities after SW update if HMI doesn't respond to `GetCapabilities` requests 
+**Given** SDL sends `GetCapabilities` requests for each interface (VR, TTS, UI, etc) to HMI
+
+**When** HMI does not send response for one ore more `GetCapabilities` requests or HMI sends errors
+
+**Then** SDL must fall back to default HMI Capabilities  
+**And** SDL must NOT persist those capabilities in cache  
```

Comment:
@KhrystynaDubovyk I am not sure about that statement. As I remember from implementation, SDL persist all received responses in cache.

Reply:
We need to double-check it for sure in this case :)

Reply:
@okandul @KhrystynaDubovyk just verified in source code - SDL actually persists all capabilities in cache on successful response for at least one of them. For example, if SDL sends `TTS.GetCapability` and `UI.GetCapability` and HMI responds to TTS capabilities with SUCCESS then SDL will save TTS part into the file. Note that UI part will be omitted in cache in that case.
Just want to note that here is not a "transaction-like" approach when SDL must persist capabilities only if all responses were successful and vise versa.

Reply:
this line in proposal made me think SDL must NOT persist those capabilities in cache
https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0249-Persisting-HMI-Capabilities-specific-to-headunit.md#hmi-capabilities-persistence-after-hmi-sw-update 
> In case HMI does not send response for GetCapabilities requests or if HMI sends error, SDL Core should fall back to default HMI Capabilities and set IsHMICooperating to true. SDL Core should not persist those capabilities in cache.

again, I might have misunderstood the proposal. 
please, confirm that it should be written `SDL must persist those capabilities in cache` according to the current implementation

Reply:
@KhrystynaDubovyk I see. Agree, this line is really confusing. So I would say that SDL must fall back to default HMI capabilities for this particular request in case when HMI sends error OR does not respond at all. Additionally, SDL should not save defaulted capabilities in cache file. However, if SDL received response from HMI for this particular request, it should use received capabilities AND save that particular response into the cache file.
Currently SDL implemented in that way